### PR TITLE
Add BMP viewer example

### PR DESCRIPTION
**Pull request by rfuest**
This PR adds a BMP viewer example. I did use a similar example during the development of `tinyqoi` and it proved to be very useful.

Unfortunately it also showed that there are more issues with this crate :disappointed:. For example, trying to view a `Rgb565` as a `Rgb555` or vice versa works without an error, but results in wrong color. And `DynamicBmp` doesn't support the color table and display 8 BPP images as `Gray8`, which isn't correct.

### CODE REVIEW

**Comment by rfuest** [comment]
The CI build was failing because the old feature resolver was used. If you don't mind I would like to update to Rust 1.57.0 and edition 2021.

**Comment by jamwaffles** [comment]
> Unfortunately it also showed that there are more issues with this crate

Sadface indeed, but it's good you found them. Someone else surely would eventually anyway.

> If you don't mind I would like to update to Rust 1.57.0 and edition 2021.

I'd like to do one more non-breaking release with #19 and any `DynamicBmp`/other bugfixes in it before we bump up to 2021 if you're good with that. I've had a lot of issues at work upgrading packages where some use edition 2021 but we were stuck on an older Rust version which makes me wary about the change.

**Comment by rfuest** [comment]
Can we merge this now as part of the next breaking release?

File: examples/display.rs [hunk 1]
```diff
@@ -0,0 +1,100 @@
+//! This example displays BMP images using the embedded-graphics simulator.
+//!
+//! Usage: cargo run --example display -- COLOR_TYPE BMP_IMAGE
```
Comment (jamwaffles) [inline examples/display.rs:3]:
```suggestion
//! Basic usage: `cargo run --example display -- COLOR_TYPE BMP_IMAGE`
//!
//! More usage and arguments can be listed by running `cargo run --example display -- --help`
```

**Review by jamwaffles — APPROVED** [review]
A small nit on the example docs, otherwise lgtm.

File: examples/display.rs (hunk 1 above)
Comment (rfuest) [reply examples/display.rs]: Thanks, that's better.

**Pull request merged**